src/apps/dialogs/TraitSelect/TraitSelectDialog.svelte

```diff
@@ -0,0 +1,225 @@
+<svelte:options accessors={true} />
+
+<script>
+    import { getContext } from "svelte";
+    import { localize } from "@typhonjs-fvtt/runtime/svelte/helper";
+
+    export let traitGroups;
+    export let selected;
+    export let disabled;
+    export let selectionLimit;
+
+    // We generate a list of disabled custom traits in order to remove them from the custom
+    // text box. This ensures these traits can not be edited. However we must still check that
+    // they won't be added either.
+    let _disabledCustom = disabled.filter((trait) => isCustomTrait(trait));
+    let _disabledSelected = selected.filter((trait) =>
+        _disabledCustom.includes(trait)
+    );
+
+    // We add selected length to the selectionLimit, since selectionLimit limits how many more
+    // elements we can add to selected. This simplifies the logic of deselecting an element and
+    // adding a different element. Since now all we have to test for is
+    // that selected.length == selectionLimit.
+    if (typeof selectionLimit === "number") {
+        selectionLimit += selected.length;
+    }
+
+    const application = getContext("external").application;
+
+    function isCustomTrait(trait) {
+        for (let group of traitGroups) {
+            if (!group.traits) continue;
+
+            for (let definedTrait in group.traits) {
+                if (definedTrait === trait) return false;
+            }
+        }
+
+        return true;
+    }
+
+    function isDisabled(trait) {
+        // Disable all nonchecked, check marks if we've reached selectionLimit
+        if (
+            selectionLimit &&
+            selected.length >= selectionLimit &&
+            !selected.includes(trait)
+        ) {
+            return true;
+        }
+
+        return disabled.includes(trait);
+    }
+
+    function setTrait(trait, status) {
+        if (status) {
+            selected = [...selected, trait];
+        } else {
+            selected.findSplice((selected) => selected === trait);
+        }
+
+        // Prevent duplicate values
+        selected = [...new Set(selected)];
+        updateDisabledStatus();
+    }
+
+    function getCustom() {
+        const custom = selected.filter(
+            (trait) => isCustomTrait(trait) && !_disabledCustom.includes(trait)
+        );
+
+        return custom.join(";");
+    }
```

Comment:
With these "Custom" functions, let's call them `getCustomTrait` or `setCustomTrait` so that we're absolutely clear what these are doing. Even the comments are kinda vague as they use the word `custom` like a noun with no further elaboration.